src/main/java/duke/tasktypes/Deadlines.java

```diff
@@ -0,0 +1,200 @@
+     * @throws DateTimeParseException if the date read is invalid.
+     */
+    public boolean checkValidityOfDateFromList(String toCheck) throws DateTimeParseException {
+        boolean isValid = true;
+        try {
+            DateTimeFormatter originalFormat = DateTimeFormatter.ofPattern("MMM/d/uuuu");
+            LocalDate.parse(toCheck, originalFormat);
+        } catch (DateTimeParseException DTPE) {
+            isValid = false;
+        }
+        return isValid;
+    }
+
+    /**
+     * Function to check if the time input by the user is valid
+     * @param toCheck String representation of the time.
+     * @return boolean indicating if the time is valid.
+     * @throws DateTimeParseException if the time is invalid.
+     */
+    public boolean checkValidityOfTime(String toCheck) throws DateTimeParseException {
+        boolean isValid = true;
+        try {
+            LocalTime.parse(toCheck);
+        } catch (DateTimeParseException DTPE) {
+            isValid = false;
+        }
+        return isValid;
+    }
+
+    /**
+     * Function to format the deadline task's due date if it is a valid input.
+     * @return String representation of the due date if valid, else string representation of prompting valid input
+     * is returned.
+     */
+    public String taskDate() {
+        String toReturn = "";
+        if (this.dueDateBy == null) {
+            toReturn = "This task does not have a valid due date! Please input the date in this format: YYYY-MM-DD";
+        } else {
+            DateTimeFormatter newFormat = DateTimeFormatter.ofPattern("MMM/d/uuuu");
+            String newFormatOfDate = this.dueDateBy.format(newFormat).replace("/", " ");
+            if (validTime) {
+                toReturn = "This task is due by: " + newFormatOfDate + " " + this.dueTime.toString();
+            } else {
+                toReturn = "This task is due by: " + newFormatOfDate;
+            }
+        }
+        return toReturn;
+    }
+
+    /**
+     * Function to check, and accordingly print String representation of deadline.
+     * @return String representation of deadline task.
+     */
+    @Override
+    public String toString() {
+        String toReturn = "";
+        DateTimeFormatter newFormat = DateTimeFormatter.ofPattern("MMM/d/uuuu");
+        if (this.dueDateBy == null) {
+            String[] checkIfTrulyNull = this.endsBy.split(" ");
+            if (checkIfTrulyNull.length >= 3) {
+                String checkingInString = checkIfTrulyNull[0] + "/" + checkIfTrulyNull[1] + "/" + checkIfTrulyNull[2];
+                if (checkValidityOfDateFromList(checkingInString)) {
+                    reformat();
+                    if (checkIfTrulyNull.length == 4) {
+                        formatTimeIfValid(checkIfTrulyNull[3]);
+                    }
+                } else {
+                    System.out.println("The date is invalid!");
+                }
+            }
+        }
```

Comment:
The module wants us to avoid deep nesting so you might want to change this if time permits